ivy_tests/test_ivy/test_functional/test_core/test_linalg.py

```diff
@@ -1046,6 +1054,33 @@ def test_tensordot(*, dtype_x1_x2_axis, test_flags, backend_fw, fn_name, on_devi
     )
 
 
+# adopted from tensorly
+# https://github.com/tensorly/tensorly/blob/main/tensorly/tenalg/tests/test_batched_tensordot.py#L9
+@pytest.mark.parametrize(
+    "shape1, shape2, axes, batched_modes",
+    [
+        ([4, 2, 3, 3], [3, 4, 2, 3], [(0, 3), (1, 3)], [1, 2]),
+        ([4, 3, 3], [3, 4, 2], [], []),
+        ([4, 3, 3], [3, 4, 2], [], [(0,), (1,)]),
+    ],
+)
+def test_tensordot_with_batched_modes_parameter(
+    backend_fw, shape1, shape2, axes, batched_modes
+):
+    import ivy
+    from tensorly.tenalg.core_tenalg import tensordot
+
+    ivy.set_backend(backend_fw)
+    tensor = ivy.random_uniform(shape=shape1)
+    tensor2 = ivy.random_uniform(shape=shape2)
+    res = ivy.tensordot(tensor, tensor2, axes=axes, batched_modes=batched_modes)
+    if backend_fw == "tensorflow":
+        tensor = tensor.numpy()
+        tensor2 = tensor2.numpy()
+    true_res = tensordot(tensor, tensor2, modes=axes, batched_modes=batched_modes)
+    assert np.allclose(res, true_res)
```

Comment:
Is this test adapted from TensorLy?

Reply:
the input in pytest.mark.parametrize are from tensorly